zerver/lib/email_mirror.py

```diff
@@ -409,11 +409,18 @@ def is_forwarded(subject: str) -> bool:
 
 def process_stream_message(to: str, message: EmailMessage) -> None:
     subject_header = message.get("Subject", "")
-    subject = strip_from_subject(subject_header) or "(no topic)"
+
+    # If the subject gets stripped to the empty string, we need to set some
+    # default value for the message topic. We can't use the usual
+    # "(no topic)" as that value is not permitted if the realm enforces
+    # that all messages must have a topic.
+    subject = strip_from_subject(subject_header) or _("Email with no subject")
 
     # We don't want to reject email messages with disallowed characters in the Subject,
     # so we just remove them to make it a valid Zulip topic name.
-    subject = "".join([char for char in subject if is_character_printable(char)]) or "(no topic)"
+    subject = "".join([char for char in subject if is_character_printable(char)]) or _(
+        "Email with no subject"
+    )
```

Comment:
Maybe we could simplify this to be just `subject = subject or _("Email with no subject")` after the loop through printable characters, with the first one being an `or ""`, since this second one would be a noop given the empty string?

Reply:
Yeah makes sense, doesn't even need the `or ""`, since `strip_from_subject` cannot return `None`, so that's redundant. Fixed.